### PR TITLE
Salus integration

### PR DESCRIPTION
**Pull request by duygugencc**
# PR Description

## Summary
In this PR, we implement Terraform code to facilitate the migration of legacy data stacks, specifically the Salus datasets, to a new GCP project named **sambla-data-staging-compliance**. This project is designed for staging data with a focus on establishing a secure and efficient architecture.

The primary goal of this migration is to enable seamless integration with our dbt project while ensuring that sensitive information is handled properly throughout the process. 

## Key Changes
The following files have been changed as part of this migration:

### Terraform Files:
- `prod/modules/salus/bigquery/iam.tf`
- `prod/modules/salus/bigquery/main.tf` (New)
- `prod/modules/salus/bigquery/salus_external_table_info.txt` (New)
- `prod/modules/salus/bigquery/variables.tf` (New)
- `prod/modules/salus/gcs/data_transfer_job.tf` (New)
- `prod/modules/salus/gcs/iam.tf` (New)
- `prod/modules/salus/gcs/main.tf` (New)
- `prod/modules/salus/gcs/variables.tf` (New)

### Schema Files:
Note: Schema files have been updated to apply policy tags to downstream models (incremental tables) and manage their integration efficiently.

### New Directory Created:
- `de-ingestion-staging-layer/prod/external_table_service`  
  This new directory contains scripts that facilitate the export and update of table schemas, supporting the smooth integration of the migrated data with downstream systems.

## Objective
This PR focuses on the migration process and the application of policy tags to the incremental downstream models. By implementing these changes, we aim to:
1. Enable secure and efficient data migration of salus to the new GCP project.
2. Ensure compatibility with the dbt project.
3. Maintain proper governance and data security through policy tag application.

## Next Steps
- Ensure that the migration is tested in the staging environment.
- Verify that the new schema definitions are successfully integrated and accessible.
- Monitor the deployment for any issues related to data access or security.

## Related Work
- Policy tags have been applied to the downstream models in the schema files to ensure compliance with data governance.

### CODE REVIEW

File: prod/external_table_service/create_tables.py
Comment (Aruldharanikumar) [inline prod/external_table_service/create_tables.py:1]: 1. Can we move the example usage block to the yaml file basically as the config?
2. Also can we change the local path here to make it dynamic?

File: prod/modules/salus/bigquery/iam.tf
Comment (Aruldharanikumar) [inline prod/modules/salus/bigquery/iam.tf:1]: Just for future scoping, can we change the actual users to group?

File: prod/modules/salus/bigquery/main.tf
Comment (Aruldharanikumar) [inline prod/modules/salus/bigquery/main.tf:1]: can we adjust tables_with_partitioning dynamically retrieve it from table_mappings by adding something like a flag while we generate prod/modules/salus/bigquery/salus_external_table_info.txt? Just a suggestion or we can leave this as it is.

File: prod/policy_tags_service/config.yaml
Comment (Aruldharanikumar) [inline prod/policy_tags_service/config.yaml:1]: @Aruldharanikumar  change the local path to dynamic through secrets in the policy tag task

**Review by Aruldharanikumar — COMMENTED** [review]
Overall It looks great! I will confirm if the terraform plan runs without any errors.

File: prod/modules/salus/bigquery/iam.tf
Comment (duygugencc) [reply prod/modules/salus/bigquery/iam.tf:1]: Yes, we can do that! Let's consider it for the next sprints!

File: prod/policy_tags_service/config.yaml
Comment (duygugencc) [reply prod/policy_tags_service/config.yaml:1]: Yes, i agree!

File: prod/modules/salus/bigquery/iam.tf [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  service_account_id = google_service_account.sa_editor.name
+  role               = "roles/iam.serviceAccountUser"  # Grant permission to use the service account
+
+  members = [
+    "user:adam.svenson@samblagroup.com",
+    "user:aruldharani.kumar@samblagroup.com",
+    "user:duygu.genc@samblagroup.com",
+  ]
+}
+
```
[the remark this answers is not in the document]
Comment (duygugencc) [reply prod/modules/salus/bigquery/iam.tf]: Done!

**Pull request merged**